enterprise data consent tests

kashifch opened the pull request:
@brittneyexline @zubair-arbi 

Please review

zubair-arbi commented on regression/pages/enterprise/enterprise_course_enrollment_page.py:
```diff
@@ -101,3 +101,15 @@ def get_course_detail_body(self):
         ).text
         return dict(zip(detail_title, detail_value))
+
+    def get_data_consent_warning(self):
```
`get_data_sharing_consent_warning`

zubair-arbi commented on regression/pages/enterprise/enterprise_course_enrollment_page.py:
```diff
@@ -101,3 +101,15 @@ def get_course_detail_body(self):
+
+    def get_data_consent_warning(self):
+        """
+        Return warning shown on rejecting data consent
```
`Returns warning shown on declining the data sharing consent.`

zubair-arbi commented on regression/pages/enterprise/enterprise_course_enrollment_page.py:
```diff
@@ -101,3 +101,15 @@ def get_course_detail_body(self):
+        """
+        return self.q(css=".alert.warning>strong").text[0]
+
+    def continue_to_data_consent_page(self):
```
`go_to_data_sharing_consent_page`

zubair-arbi commented on regression/pages/enterprise/enterprise_course_enrollment_page.py:
```diff
@@ -101,3 +101,15 @@ def get_course_detail_body(self):
+
+    def continue_to_data_consent_page(self):
+        """
+        Go to data consent page
```
nit: `Go to data sharing consent page.`

zubair-arbi commented on regression/pages/enterprise/enterprise_data_consent_page.py:
```diff
@@ -0,0 +1,85 @@
+"""
+Enterprise Data Consent page
```
`Enterprise data sharing consent page.`

zubair-arbi commented on regression/pages/enterprise/enterprise_data_consent_page.py:
```diff
@@ -0,0 +1,85 @@
+)
+
+
+class EnterpriseDataConsentPage(PageObject):
```
`DataSharingConsentPage` is just fine.

zubair-arbi commented on regression/pages/enterprise/enterprise_data_consent_page.py:
```diff
@@ -0,0 +1,85 @@
+        )
+        self.q(css="#consent-button").click()
+
+    def decline_data_consent(self):
```
```python
def decline_data_sharing_consent(self):
    """
    Check decline checkbox on data enterprise sharing consent page.
    """
```

zubair-arbi commented on regression/tests/enterprise/test_enterprise_data_consent.py:
```diff
@@ -0,0 +1,126 @@
+        # is displayed there
+        self.enterprise_course_enrollment.wait_for_page()
+        self.assertDictEqual(
+            self.ENT_COURSE_TITLE,
```
We are doing all the login stuff and going to enterprise course enrollment page in lot of other places. I would suggest creating a helper method like `go_to_enterprise_course_enrollment_page` and move all this logic there.

zubair-arbi commented on regression/tests/enterprise/test_enterprise_enrollment_page.py:
```diff
@@ -9,7 +9,7 @@
 class TestEnterpriseEnrollmentPage(EnterpriseTestBase):
     """
-    Test Enterprise Login
+    Test Enterprise Enrollment page
```
Just for info that there are two enrollment pages in enterprise one is `enterprise course enrollment page` and other one is `enterprise program enrollment page`.

Since you have this class `TestEnterpriseEnrollmentPage` so I believe it will wrap all the tests related to both pages if not then I would suggest changing this class name to `TestEnterpriseCourseEnrollmentPage`.

zubair-arbi commented on regression/tests/enterprise/test_enterprise_data_consent.py:
```diff
@@ -0,0 +1,126 @@
+        # is displayed there
+        self.enterprise_course_enrollment.wait_for_page()
+        self.assertDictEqual(
+            self.ENT_COURSE_TITLE,
```
Please use util method `go_to_enterprise_course_enrollment_page` to get to enterprise course enrollment page as suggested above.

kashifch commented:
@zubair-arbi thanks for the suggestions, I have changed the nomenclature for all tests and pages and replaced enterprise with ent, so we can choose meaningful names without making these excessively long

kashifch commented:
@zubair-arbi 
the names were changed because some of the function names exceed pep8 specs, sorry for the extra work

zubair-arbi commented:
@edx/testeng Could someone of you please have a look on the changes as well?
@brittneyexline It would be great if you could also review.
